query/select_test.go

```diff
@@ -17,6 +17,808 @@ import (
+			q:    `SELECT time_weighted_average(value) FROM cpu WHERE time >= '1970-01-01T00:00:00Z' AND time < '1970-01-01T00:00:14Z' GROUP BY time(2s)'`,
+			typ:  influxql.Float,
+			itrs: []query.Iterator{
+				&FloatIterator{Points: []query.FloatPoint{
+					{Name: "cpu", Time: 1 * Second, Value: 4},
+					{Name: "cpu", Time: 3 * Second, Value: 2},
+					{Name: "cpu", Time: 5 * Second, Value: 3},
+					{Name: "cpu", Time: 7 * Second, Value: 4},
+					{Name: "cpu", Time: 9 * Second, Value: 5},
+					{Name: "cpu", Time: 11 * Second, Value: 6},
+					{Name: "cpu", Time: 13 * Second, Value: 7},
+					{Name: "cpu", Time: 14 * Second, Value: 7},
+				}},
+			},
+			rows: []query.Row{
+				{Time: 0 * Second, Series: query.Series{Name: "cpu"}, Values: []interface{}{float64(4)}},
+				{Time: 2 * Second, Series: query.Series{Name: "cpu"}, Values: []interface{}{float64(3)}},
+				{Time: 4 * Second, Series: query.Series{Name: "cpu"}, Values: []interface{}{float64(2.5)}},
+				{Time: 6 * Second, Series: query.Series{Name: "cpu"}, Values: []interface{}{float64(3.5)}},
+				{Time: 8 * Second, Series: query.Series{Name: "cpu"}, Values: []interface{}{float64(4.5)}},
+				{Time: 10 * Second, Series: query.Series{Name: "cpu"}, Values: []interface{}{float64(5.5)}},
+				{Time: 12 * Second, Series: query.Series{Name: "cpu"}, Values: []interface{}{float64(6.5)}},
+				{Time: 14 * Second, Series: query.Series{Name: "cpu"}, Values: []interface{}{float64(7)}},
+			},
+		},
+		{
+			name: "TimeWeightedAverage_Float_NoGrouping_SingleValueAtMiddle",
+			q:    `SELECT time_weighted_average(value) FROM cpu WHERE time >= '1970-01-01T00:00:00Z' AND time < '1970-01-01T00:00:14Z'`,
+			typ:  influxql.Float,
+			itrs: []query.Iterator{
+				&FloatIterator{Points: []query.FloatPoint{
+					{Name: "cpu", Time: 7 * Second, Value: 2},
+				}},
+			},
+			rows: []query.Row{
+				{Time: 0 * Second, Series: query.Series{Name: "cpu"}, Values: []interface{}{float64(2)}},
+			},
+		},
+		{
+			name: "TimeWeightedAverage_Float_NoGrouping_SingleValueAtBeginning",
+			q:    `SELECT time_weighted_average(value) FROM cpu WHERE time >= '1970-01-01T00:00:00Z' AND time < '1970-01-01T00:00:13Z'`,
+			typ:  influxql.Float,
+			itrs: []query.Iterator{
+				&FloatIterator{Points: []query.FloatPoint{
+					{Name: "cpu", Time: 0 * Second, Value: 2},
+				}},
+			},
+			rows: []query.Row{
+				{Time: 0 * Second, Series: query.Series{Name: "cpu"}, Values: []interface{}{float64(2)}},
+			},
+		},
+		{
+			name: "TimeWeightedAverage_Float_ValuesOnlyBeforeRequestedWindow_NoGrouping",
+			q:    `SELECT time_weighted_average(value) FROM cpu WHERE time >= '1970-01-01T00:00:02Z' AND time < '1970-01-01T00:00:10Z'`,
+			typ:  influxql.Float,
+			itrs: []query.Iterator{
+				&FloatIterator{Points: []query.FloatPoint{	
+					{Name: "cpu", Time: 0 * Second, Value: 2},				
+				}},
+			},
+			rows: []query.Row(nil),
+		},		
+		{
+			name: "TimeWeightedAverage_Float_NoGrouping_ValuesOnlyAfterRequestedWindow",
```

Comment:
Sample doesn't fit with test name. Should be : 
`{Name: "cpu", Time: 14 * Second, Value: 2},`